sdk/macro/src/lib.rs

```diff
@@ -22,7 +22,7 @@ impl Parse for Id {
                 .into_vec()
                 .map_err(|_| syn::Error::new_spanned(&id_literal, "failed to decode base58 id"))?;
             let id_array = <[u8; 32]>::try_from(<&[u8]>::clone(&&id_vec[..]))
-                .map_err(|_| syn::Error::new_spanned(&id_literal, "id is not 32 bytes long"))?;
+                .map_err(|_| syn::Error::new_spanned(&id_literal, format!("id is not 32 bytes long: len={}", id_vec.len())))?;
```

Comment:
Oh, I guess you just copied [my comment's string](https://github.com/solana-labs/solana/pull/9815#discussion_r418184631)... Sorry for being lazy. ;)